File: rb/features.rb
# RefugeRestrooms SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module RefugeRestroomsFeatures
  def self.make_feature(name)
    case name
    when "base"
      RefugeRestroomsBaseFeature.new
    when "ratelimit"
      RefugeRestroomsRatelimitFeature.new
    when "retry"
      RefugeRestroomsRetryFeature.new
    when "test"
      RefugeRestroomsTestFeature.new
    when "timeout"
      RefugeRestroomsTimeoutFeature.new
    else
      RefugeRestroomsBaseFeature.new
    end
  end
end

File: py/features.py
# RefugeRestrooms SDK feature factory

from feature.base_feature import RefugeRestroomsBaseFeature
from feature.test_feature import RefugeRestroomsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RefugeRestroomsBaseFeature(),
        "test": lambda: RefugeRestroomsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
